src/ui/server/controllers/insurance/policy/pre-credit-period/index.ts

```diff
@@ -107,6 +112,29 @@ export const post = async (req: Request, res: Response) => {
   }
 
   const { referenceNumber } = req.params;
+  const refNumber = Number(referenceNumber);
+
+  const payload = constructPayload(req.body, FIELD_IDS);
+  const sanitisedData = sanitiseData(payload);
+
+  const validationErrors = generateValidationErrors(payload);
+
+  if (validationErrors) {
+    return res.render(TEMPLATE, {
+      ...insuranceCorePageVariables({
+        PAGE_CONTENT_STRINGS,
+        BACK_LINK: req.headers.referer,
+        HTML_FLAGS,
+      }),
+      ...pageVariables(refNumber),
+      FIELD_ID: NEED_PRE_CREDIT_PERIOD,
```

Comment:
is there a reason that these are here and not in pageVariables?

Reply:
Excellent spot, have moved into page variables